tests/blobcache.bats

```diff
@@ -55,6 +55,37 @@ function _check_matches() {
         expect_output --from="$unmatched" "$5"  "$6 should not match"
 }
 
+# Integration test for https://github.com/containers/image/pull/1645
+@test "blobcache: blobs must be reused when pushing across registry" {
+	start_registry
+	run_buildah login --tls-verify=false --authfile ${TEST_SCRATCH_DIR}/test.auth --username testuser --password testpassword localhost:${REGISTRY_PORT}
+	run_buildah build $WITH_POLICY_JSON -t exp -f $BUDFILES/simple-multi-step/Containerfile
+	run_buildah push $WITH_POLICY_JSON --format docker exp dir:${TEST_SCRATCH_DIR}/exp
+	run_buildah --log-level debug push --tls-verify=false --authfile ${TEST_SCRATCH_DIR}/test.auth exp docker://localhost:${REGISTRY_PORT}/test
```

Comment:
Pedantically, the compression algorithm is not even promised to be deterministic: one compression to `dir:…` and another to `docker://` can create different compressed representations. So I think this still has the same problem, although in an even-rarer-to-hit variant.

Reply:
Can it be made more certain if i add `--compression gzip` explicitly ?

Reply:
That doesn’t help:
- Drop BIC
- `buildah push foo docker://bar` (compressing)
- Drop BIC
- `buildah push foo docker://bar` (compressing again)

is _not_ required to create bit-identical representations. (Without the second BIC drop, we would reuse blobs and _probably_ use identical representations — depending on which compressed variant the BIC chooses.)

That holds regardless of algorithm: AFAIK, none of the promise to compress the same input to the same output, even in the same version (though I wasn’t looking for such a promise very hard).

Reply:
Now moving image directly from repo to registry and then clearing BIC and performing push.